lib/gateway/build-gateway.js

```diff
@@ -357,7 +357,11 @@ function defaultErrorHandler (error, service) {
 async function buildGateway (serviceMap, gatewayOpts, app, lruGatewayResolvers) {
   const { services, errorHandler = defaultErrorHandler } = gatewayOpts
 
-  await buildServiceMap(serviceMap, services, errorHandler, app.log)
+  if (services instanceof Function) {
+    await buildServiceMap(serviceMap, await services(), errorHandler, app.log)
```

Comment:
what happens in case of rejection? maybe wrap `await services()` in a function with trycatch would help with error handling

Reply:
Rejection is handled inside `buildServiceMap` as it is done in the rest of the code